Description: CVE-2011-0713 Wireshark: heap-based buffer overflow
Forwarded: http://anonsvn.wireshark.org/viewvc/viewvc.cgi?view=rev&revision=35953
Author: Gerald Combs

From https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2011-0713 :

A heap-based buffer overflow was found in the way Wireshark
processes signalling traces generated by Gammu (www.gammu.org)
from Nokia DCT3 phones in Netmonitor mode.

An attacker could use this flaw to cause wireshark executable
to crash or, potentially, execute arbitrary code with the privileges
of the user running wireshark, if the local user opened a
specially-crafted capture file.


Index: trunk/wiretap/dct3trace.c
===================================================================
--- trunk/wiretap/dct3trace.c	(revision 35952)
+++ trunk/wiretap/dct3trace.c	(revision 35953)
@@ -392,6 +392,13 @@
 		return FALSE;
 	}
 
+	if( buf_len > MAX_PACKET_LEN)
+	{
+		*err = WTAP_ERR_BAD_RECORD;
+		*err_info = g_strdup_printf("dct3trace: record length %d too long", buf_len);
+		return FALSE;
+	}
+
 	memcpy( pd, buf, buf_len );
 	return TRUE;
 }
